### PR TITLE
CORE-7146 TransactionMetadata tests

### PR DESCRIPTION
**Pull request by vlajos**
Some invalid transaction metadata edge case tests around the WireTransctionFactoryImpl.

### CODE REVIEW

File: libs/ledger/ledger-common-data/src/test/kotlin/net/corda/ledger/common/data/transaction/factory/WireTransactionFactoryImplTest.kt [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+    fun `Can create very simple WireTransaction`() {
+        wireTransactionFactory.create(
+            listOf(
+                listOf(canonicalJson.toByteArray()),
+            ), privacySalt
+        )
+    }
+
+    @Test
+    fun `Trying to create WireTransaction with non-canonical metadata throws`() {
```
Comment (lankydan) [inline libs/ledger/ledger-common-data/src/test/kotlin/net/corda/ledger/common/data/transaction/factory/WireTransactionFactoryImplTest.kt]: Can we change these to `Creating a...` rather than `Trying to create`

**Review by lankydan — DISMISSED** [review]
Tests look fine but left a comment on the test names. You can change them or leave them, not a big deal.

File: libs/ledger/ledger-common-data/src/test/kotlin/net/corda/ledger/common/data/transaction/factory/WireTransactionFactoryImplTest.kt (hunk 1 above)
Comment (vlajos) [reply libs/ledger/ledger-common-data/src/test/kotlin/net/corda/ledger/common/data/transaction/factory/WireTransactionFactoryImplTest.kt]: I have just updated them.

**Review by lankydan — APPROVED** [review]
